#Publish all joint states -----------------------------------
joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 100

joint_motor_controller:
    type: velocity_controllers/JointGroupVelocityController
    publish_rate: 100
    joints: 
        - shoulder_pan_joint
        - shoulder_lift_joint
        - elbow_joint
        - wrist_1_joint
        - wrist_2_joint
        - wrist_3_joint
gains:
    shoulder_pan_joint: {p: 10000, i: 1, d: 1000}
    shoulder_lift_joint: {p: 10000, i: 1, d: 1000, i_clamp_min: -100.0, i_clamp_max: 100.0}
    elbow_joint: {p: 10000, i: 1, d: 1000}
    wrist_1_joint: {p: 10000, i: 1, d: 1000}
    wrist_2_joint: {p: 10000, i: 1, d: 1000}
    wrist_2_joint: {p: 10000, i: 1, d: 1000}


#   Vel Position Controllers ---------------------------------------
shoulder_pan_joint_position_controller:
    type: velocity_controllers/JointPositionController
    joint: shoulder_pan_joint
    pid: {p: 100.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
shoulder_lift_joint_position_controller:
    type: velocity_controllers/JointPositionController
    joint: shoulder_lift_joint
    pid: {p: 100.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
elbow_joint_position_controller:
    type: velocity_controllers/JointPositionController
    joint: elbow_joint
    pid: {p: 100.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
wrist_1_joint_position_controller:
    type: velocity_controllers/JointPositionController
    joint: wrist_1_joint
    pid: {p: 100.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
wrist_2_joint_position_controller:
    type: velocity_controllers/JointPositionController
    joint: wrist_2_joint
    pid: {p: 100.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
wrist_3_joint_position_controller:
    type: velocity_controllers/JointPositionController
    joint: wrist_3_joint
    pid: {p: 100.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
  
  # #Effort controllers
  # shoulder_pan_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: shoulder_pan_joint
  #   # pid: {p: 100.0, i: 0.0, d: 100.00}
  # shoulder_lift_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: shoulder_lift_joint
  #   pid: {p: 0.0, i: 0.0, d: 0.0}
  # elbow_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: elbow_joint
  #   pid: {p: 0.0, i: 0.0, d: 0.0}
  # wrist_1_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: wrist_1_joint
  #   pid: {p: 0.0, i: 0.0, d: 0.0}
  # wrist_2_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: wrist_2_joint
  #   pid: {p: 0.0, i: 0.0, d: 0.0}
  # wrist_3_joint_velocity_controller:
  #   type: effort_controllers/JointVelocityController
  #   joint: wrist_3_joint
  #   pid: {p: 0.0, i: 0.0, d: 0.0}

  #Velocity Controllers
  # shoulder_pan_joint_velocity_controller:
  #   type: velocity_controllers/JointVelocityController
  #   joint: shoulder_pan_joint
  #   pid: {p: 500.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
  # shoulder_lift_joint_velocity_controller:
  #   type: velocity_controllers/JointVelocityController
  #   joint: shoulder_lift_joint
  #   pid: {p: 10000.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
  # elbow_joint_velocity_controller:
  #   type: velocity_controllers/JointVelocityController
  #   joint: elbow_joint
  #   pid: {p: 10000.0, i: 0.01, d: 50.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
  # wrist_1_joint_velocity_controller:
  #   type: velocity_controllers/JointVelocityController
  #   joint: wrist_1_joint
  #   pid: {p: 200.0, i: 10.0, d: 20.0, i_clamp_min: -400.0, i_clamp_max: 400.0}
  # wrist_2_joint_velocity_controller:
  #   type: velocity_controllers/JointVelocityController
  #   joint: wrist_2_joint
  #   pid: {p: 100.0, i: 0.1, d: 10.0, i_clamp_min: -100.0, i_clamp_max: 100.0}
  # wrist_3_joint_velocity_controller:
  #   type: velocity_controllers/JointVelocityController
  #   joint: wrist_3_joint
  #   pid: {p: 100.0, i: 0.1, d: 10.0, i_clamp_min: -100.0, i_clamp_max: 100.0}